PHP7 branch build on Win32 VC14

Hi,
1. build v8 with using VS 2015,  everything looks OK.
2. copied the libs and headers to deps in PHP build tree   V7.0.1
3. copied the v8js branch PHP7 to ext folder in PHP build tree
4. configure  --disable-all --enable-cli --enable-cli-win32  --with-v8js
5. nmake
result 
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\INCLUDE\chrono(778): warning C4091: "__forceinline ": Ignoriert auf der linken Seite von "int", wenn keine Variable deklariert wurde (Quelldatei wird kompiliert ext\v8js\v8js.cc)
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\INCLUDE\chrono(778): error C2143: Syntaxfehler: Es fehlt ";" vor "namespace" (Quelldatei wird kompiliert ext\v8js\v8js.cc)
....
NMAKE : fatal error U1077: ""C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\cl.exe"": Rückgabe-Code "0x2"

looks like a problem with inline name space  ?
chrono
..
774		return (_Xt);
775		}
776	
777		// duration LITERALS
778	inline namespace literals {
779	inline namespace chrono_literals {
780	inline _CONST_FUN chrono::hours operator "" h(unsigned long long _Val)
..

Hi

yes this error is fixed bad now there is a other problem.
Now they include abs and also extern "C"  abs 

C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\INCLUDE\cmath(17): error C2894: Vorlagen können nicht mit "C"-Bindung deklariert werden (Quelldatei wird kompiliert ext\v8js\v8js_array_access.cc)
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\INCLUDE\cmath(35): error C2733: "abs" : Zweite C-Bindung fr berladene Funktion nicht zul„ssig (Quelldatei wird kompiliert ext\v8js\v8js_array_access.cc)
C:\Program Files (x86)\Windows Kits\10\include\10.0.10586.0\ucrt\stdlib.h(286): note: Siehe Deklaration von "abs" (Quelldatei wird kompiliert ext\v8js\v8js_array_access.cc)
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\INCLUDE\cmath(40): error C2733: "pow" : Zweite C-Bindung fr berladene Funktion nicht zul„ssig (Quelldatei wird kompiliert ext\v8js\v8js_array_access.cc)
C:\Program Files (x86)\Windows Kits\10\include\10.0.10586.0\ucrt\math.h(483): note: Siehe Deklaration von "pow" (Quelldatei wird kompiliert ext\v8js\v8js_array_access.cc)
.....

cheers
walter

--- a/php_v8js_macros.h
+++ b/php_v8js_macros.h
@@ -18,6 +18,15 @@
 #define _GLIBCXX_USE_NANOSLEEP 1
 #endif
 
+#include <chrono>
+#include <deque>
+#include <thread>
+
+#include <map>
+#include <list>
+#include <vector>
+#include <mutex>
+
 extern "C" {
 #include "php.h"
 #include "php_v8js.h"
@@ -30,15 +39,6 @@ extern "C" {
 #endif
 
 #include <v8.h>
-
-#include <chrono>
-#include <deque>
-#include <thread>
-
-#include <map>
-#include <list>
-#include <vector>
-#include <mutex>
 
 #include "v8js_class.h"
 #include "v8js_v8.h"

--- a/v8js_array_access.cc
+++ b/v8js_array_access.cc
@@ -14,6 +14,10 @@
 #include "config.h"
 #endif
 
+#include "php_v8js_macros.h"
+#include "v8js_array_access.h"
+#include "v8js_object_export.h"
+
 extern "C" {
 #include "php.h"
 #include "ext/date/php_date.h"
@@ -21,11 +25,6 @@ extern "C" {
 #include "zend_interfaces.h"
 #include "zend_closures.h"
 }
-
-#include "php_v8js_macros.h"
-#include "v8js_array_access.h"
-#include "v8js_object_export.h"
-
 
 static zval v8js_array_access_dispatch(zend_object *object, const char *method_name, int param_count,
 									   uint32_t index, zval zvalue TSRMLS_DC) /* {{{ */

--- a/v8js_class.cc
+++ b/v8js_class.cc
@@ -16,6 +16,15 @@
 #include "config.h"
 #endif
 
+#include <functional>
+#include <algorithm>
+
+#include "php_v8js_macros.h"
+#include "v8js_v8.h"
+#include "v8js_exceptions.h"
+#include "v8js_v8object_class.h"
+#include "v8js_timer.h"
+
 extern "C" {
 #include "php.h"
 #include "ext/date/php_date.h"
@@ -25,15 +34,6 @@ extern "C" {
 #include "ext/spl/spl_exceptions.h"
 #include "zend_exceptions.h"
 }
-
-#include "php_v8js_macros.h"
-#include "v8js_v8.h"
-#include "v8js_exceptions.h"
-#include "v8js_v8object_class.h"
-#include "v8js_timer.h"
-
-#include <functional>
-#include <algorithm>
 
 #define PHP_V8JS_SCRIPT_RES_NAME "V8Js script"
 

--- a/v8js_commonjs.cc
+++ b/v8js_commonjs.cc
@@ -15,12 +15,12 @@
 #include "config.h"
 #endif
 
+#include "php_v8js_macros.h"
+
 extern "C" {
 #include "php.h"
 #include "zend_exceptions.h"
 }
-
-#include "php_v8js_macros.h"
 
 static void v8js_commonjs_split_terms(const char *identifier, std::vector<char *> &terms)
 {

--- a/v8js_convert.cc
+++ b/v8js_convert.cc
@@ -15,6 +15,14 @@
 #include "config.h"
 #endif
 
+#include <stdexcept>
+#include <limits>
+
+#include "php_v8js_macros.h"
+#include "v8js_object_export.h"
+#include "v8js_v8object_class.h"
+#include "v8js_v8.h"
+
 extern "C" {
 #include "php.h"
 #include "ext/date/php_date.h"
@@ -22,14 +30,6 @@ extern "C" {
 #include "zend_interfaces.h"
 #include "zend_closures.h"
 }
-
-#include "php_v8js_macros.h"
-#include "v8js_object_export.h"
-#include "v8js_v8object_class.h"
-#include "v8js_v8.h"
-
-#include <stdexcept>
-#include <limits>
 
 static int v8js_is_assoc_array(HashTable *myht TSRMLS_DC) /* {{{ */
 {

--- a/v8js_exceptions.cc
+++ b/v8js_exceptions.cc
@@ -16,8 +16,9 @@
 #include "config.h"
 #endif
 
+#include "php_v8js_macros.h"
+
 extern "C" {
-#include "php.h"
 #include "ext/date/php_date.h"
 #include "ext/standard/php_string.h"
 #include "zend_interfaces.h"
@@ -25,9 +26,6 @@ extern "C" {
 #include "ext/spl/spl_exceptions.h"
 #include "zend_exceptions.h"
 }
-
-#include "php_v8js_macros.h"
-
 
 /* {{{ Class Entries */
 zend_class_entry *php_ce_v8js_exception;

--- a/v8js_object_export.cc
+++ b/v8js_object_export.cc
@@ -15,19 +15,18 @@
 #include "config.h"
 #endif
 
+#include "php_v8js_macros.h"
+#include "v8js_array_access.h"
+#include "v8js_object_export.h"
+#include "v8js_v8object_class.h"
+
 extern "C" {
-#include "php.h"
 #include "ext/date/php_date.h"
 #include "ext/standard/php_string.h"
 #include "zend_interfaces.h"
 #include "zend_closures.h"
 #include "zend_exceptions.h"
 }
-
-#include "php_v8js_macros.h"
-#include "v8js_array_access.h"
-#include "v8js_object_export.h"
-#include "v8js_v8object_class.h"
 
 static void v8js_weak_object_callback(const v8::WeakCallbackData<v8::Object, zend_object> &data);
 

--- a/v8js_timer.cc
+++ b/v8js_timer.cc
@@ -15,7 +15,9 @@
 #include "config.h"
 #endif
 
+#ifdef _WIN32
 #include <concrt.h>
+#endif
 
 #include "php_v8js_macros.h"
 #include "v8js_v8.h"

--- a/v8js_timer.cc
+++ b/v8js_timer.cc
@@ -15,8 +15,14 @@
 #include "config.h"
 #endif
 
+#include <concrt.h>
+
+#include "php_v8js_macros.h"
+#include "v8js_v8.h"
+#include "v8js_exceptions.h"
+#include "v8js_timer.h"
+
 extern "C" {
-#include "php.h"
 #include "ext/date/php_date.h"
 #include "ext/standard/php_string.h"
 #include "zend_interfaces.h"
@@ -24,11 +30,6 @@ extern "C" {
 #include "ext/spl/spl_exceptions.h"
 #include "zend_exceptions.h"
 }
-
-#include "php_v8js_macros.h"
-#include "v8js_v8.h"
-#include "v8js_exceptions.h"
-#include "v8js_timer.h"
 
 static void v8js_timer_interrupt_handler(v8::Isolate *isolate, void *data) { /* {{{ */
 	zend_v8js_globals *globals = static_cast<zend_v8js_globals *>(data);

--- a/v8js_v8.cc
+++ b/v8js_v8.cc
@@ -16,8 +16,12 @@
 #include "config.h"
 #endif
 
+#include "php_v8js_macros.h"
+#include "v8js_v8.h"
+#include "v8js_timer.h"
+#include "v8js_exceptions.h"
+
 extern "C" {
-#include "php.h"
 #include "ext/date/php_date.h"
 #include "ext/standard/php_string.h"
 #include "zend_interfaces.h"
@@ -28,11 +32,6 @@ extern "C" {
 #if !defined(_WIN32) && PHP_V8_API_VERSION >= 3029036
 #include <libplatform/libplatform.h>
 #endif
-
-#include "php_v8js_macros.h"
-#include "v8js_v8.h"
-#include "v8js_timer.h"
-#include "v8js_exceptions.h"
 
 void v8js_v8_init(TSRMLS_D) /* {{{ */
 {

--- a/v8js_v8object_class.cc
+++ b/v8js_v8object_class.cc
@@ -15,8 +15,12 @@
 #include "config.h"
 #endif
 
+#include "php_v8js_macros.h"
+#include "v8js_exceptions.h"
+#include "v8js_v8.h"
+#include "v8js_v8object_class.h"
+
 extern "C" {
-#include "php.h"
 #include "ext/date/php_date.h"
 #include "ext/standard/php_string.h"
 #include "zend_interfaces.h"
@@ -24,11 +28,6 @@ extern "C" {
 #include "ext/spl/spl_exceptions.h"
 #include "zend_exceptions.h"
 }
-
-#include "php_v8js_macros.h"
-#include "v8js_exceptions.h"
-#include "v8js_v8.h"
-#include "v8js_v8object_class.h"
 
 /* {{{ Class Entries */
 zend_class_entry *php_ce_v8object;

--- a/v8js_variables.cc
+++ b/v8js_variables.cc
@@ -17,13 +17,9 @@
 #include "config.h"
 #endif
 
-extern "C" {
-#include "php.h"
-}
+#include <string>
 
 #include "php_v8js_macros.h"
-#include <v8.h>
-#include <string>
 
 static void v8js_fetch_php_variable(v8::Local<v8::String> name, const v8::PropertyCallbackInfo<v8::Value>& info) /* {{{ */
 {

